File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    void checkPalindrome(string &s, int start, int end) {
        if (start > end)
            return;
        swap(s[start], s[end]);
        checkPalindrome(s, start + 1, end - 1);
    }
    
    bool isPalindrome(string s) {
        string new_s;
        for (int i = 0; i < s.size(); i++) {
            int c = int(s[i]);
            if ((c >= 65 && c <= 90) || (c >= 97 && c <= 122) || (c >= 48 && c <= 57)) {
                if (c >= 65 && c <= 90)
                    new_s.push_back(int(s[i]) + 32);
                else
                    new_s.push_back(s[i]);
            }
        }
        // cout << new_s << endl;
        string temp = new_s;
        checkPalindrome(new_s, 0, new_s.size() - 1);
        if (new_s == temp)
            return 1;
        return 0;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        cleaned = ""
        for i in range(len(s)):
            if (ord(s[i]) >= 97 and ord(s[i]) <= 122) or (ord(s[i]) >= 65 and ord(s[i]) <= 90) or (ord(s[i]) >= 48 and ord(s[i]) <= 57):
                cleaned += s[i].lower()
        if cleaned[::-1] == cleaned:
            return True
        return False
            